pkg/spiffe/spiffe.go

```diff
@@ -42,10 +47,7 @@ func GenSpiffeURI(ns, serviceAccount string) (string, error) {
 		err = fmt.Errorf(
 			"namespace or service account can't be empty ns=%v serviceAccount=%v", ns, serviceAccount)
 	}
-
-	// replace specifial character in spiffe
-	trustDomain = strings.Replace(trustDomain, "@", ".", -1)
-	return fmt.Sprintf(Scheme+"://%s/ns/%s/sa/%s", trustDomain, ns, serviceAccount), err
+	return uriPrefix + trustDomain + "/ns/" + ns + "/sa/" + serviceAccount, err
```

Comment:
Not a problem with your checkin in particular, but do the ns and serviceAccount need to be escaped?

Reply:
I'm not sure.

@quanjielin @wattli may have more